filesystems/blob/src/main/scala/cromwell/filesystems/blob/BlobPathBuilderFactory.scala

```diff
@@ -1,43 +1,89 @@
 package cromwell.filesystems.blob
 
 import akka.actor.ActorSystem
+import com.azure.core.credential.AzureSasCredential
 import com.azure.core.management.AzureEnvironment
 import com.azure.core.management.profile.AzureProfile
 import com.azure.identity.DefaultAzureCredentialBuilder
 import com.azure.resourcemanager.AzureResourceManager
 import com.azure.storage.blob.BlobContainerClientBuilder
+import com.azure.storage.blob.nio.AzureFileSystem
 import com.azure.storage.blob.sas.{BlobContainerSasPermission, BlobServiceSasSignatureValues}
 import com.azure.storage.common.StorageSharedKeyCredential
 import com.typesafe.config.Config
 import cromwell.core.WorkflowOptions
 import cromwell.core.path.PathBuilderFactory
 import net.ceedubs.ficus.Ficus._
 
-import java.time.OffsetDateTime
-import scala.concurrent.ExecutionContext
-import scala.concurrent.Future
+import java.net.URI
+import java.nio.file.{FileSystem, FileSystemNotFoundException, FileSystems}
+import java.time.temporal.ChronoUnit
+import java.time.{Duration, OffsetDateTime}
+import scala.concurrent.{ExecutionContext, Future}
 import scala.jdk.CollectionConverters._
+import scala.util.{Failure, Try}
 
 final case class BlobFileSystemConfig(config: Config)
 final case class BlobPathBuilderFactory(globalConfig: Config, instanceConfig: Config, singletonConfig: BlobFileSystemConfig) extends PathBuilderFactory {
```

Comment:
We have a LOT of important string identifiers floating around. Not necessarily in scope for this PR, but we should consider replacing them with simple case classes so that the compiler can check that we're passing them around correctly. Otherwise it's too easy to swap their ordering in a method call and introduce a very hard-to-find bug.

```
case class BlobContainerName(value: String)
case class StorageAccountName(value: String)
case class WorkspaceId(value: String)
```
...and so on.

Reply:
I think I will give this a try, It will likely help me get the tests right without accidentally swapping strings